main.nf

```diff
@@ -437,10 +516,34 @@ if( !params.bwa_meth_index && params.aligner == 'bwameth' ){
     }
 }
 
+/*
+ * PREPROCESSING - Build bwa-biscuit
+ */
+if(!params.bwa_biscuit_index && params.aligner == 'biscuit' ){
+    process makeBwaBISCUITIndex {
+        tag "$fasta"
+        publishDir path: "${params.outdir}/reference_genome", saveAs: { params.save_reference ? it : null }, mode: params.publish_dir_mode
+
+        input:
+        file fasta from ch_fasta_for_makeBwaMemIndex
+
+        output:
+        file "${fasta}*" into ch_bwa_index_for_biscuit
+
+        script:
```

Comment:
you could simplify this by using `stageAs` for the input channel. See [here](https://github.com/nf-core/modules/blob/7996a461786c33fb493d7356c8810adc436aca5f/software/bismark/genome_preparation/main.nf#L22) for example

Reply:
You mean something like this?
```
 input:
        file fasta stageAs "${fasta}*" from ch_fasta_for_makeBwaMemIndex

        output:
        file fasta into ch_bwa_index_for_biscuit
```

So the same should be applied to the `makeBwaMemIndex` process

Reply:
Yes good catch! For consistency it would be best to do the same for `makeBwaMemIndex` and `makeBismarkIndex` as well, you're right.

Reply:
The above did not work, maybe I'm doing something wrong.
Maybe I'll just wait for DSL2 with this, as you can see below

Reply:
Ah now I see why, `stageAs` requires the `path` qualifier instead of `file` but the latter is used throughout the pipeline.
No worries, let's just make the switch to `path` with DSL2 :+1: